### DISCUSSION TITLE
tpi - problemas con strcmp y strcasecmp

### DISCUSSION THREAD

**Discussion by rinrot** [INGCOM-UNRN-P1/cursada-2023#233] [Preguntas y respuestas]
Hola! estoy intentando que un bucle while se cierre cuando una comparacion de caracteres con strcmp de 0 (osea que sean iguales),con una siendo una cadena fija y la otra va cambiando, pero por alguna razon aunque las 2 sean iguales el valor que da la funcion sigue siendo -1 (osea, que el valor str1 es menor al valor str2). cual podria ser el problema? 

la funcion es asi: `strcmp(nombre_archivo, "data\\habitacion_1\\habitacion_hub.txt")`

siempre da -1, incluso cuando supuestamente son iguales.

**Comment by martinvilu** [comment]
¿Qué tan idénticos son?, ¿en dónde está el `\0` en ambas cadenas? ¿Probaste con la función creada para cadena segura?

**Reply by rinrot** [reply]
probe haciendolo asi `strcmp(nombre_archivo, "data\\habitacion_1\\habitacion_hub.txt\0")` y tampoco funcionaba.

**Reply by martinvilu** [reply]
Sí, pero mira en detalle como son ambas cadenas

**Reply by rinrot** [reply]
Ya lo hice. cuando la imprimo la cadena es exactamente igual. tampoco me olvide de ponerle el `\0` al final

**Reply by rinrot** [reply]
probe usar strcasecmp y tampoco funciono. tambien probe almacenando el string en una variable const char y tampoco funciono.

**Reply by martinvilu** [reply]
Puede ser la forma en la que están armadas las rutas… si es con `\` o con `/`.

**Reply by rinrot** [reply]
> Puede ser la forma en la que están armadas las rutas… si es con `\` o con `/`.

Tampoco es el problema. Podes ver lo que deje en el repositorio para ver cual es el problema? De verdad que no se como solucionarlo.
